twifbeddit-app/backend/src/users.js

```diff
@@ -29,3 +29,60 @@ exports.POST = async (_, event) => {
 
     };
 };
+
+// TODO: Test this. Gets the topics that the user is following.
+exports.get_topics_that_the_user_is_following = async (_, event) => {
+    if (event.queryStringParameters != null)
+        console.error("Parameters currently ignored");
+    const username = await get_user_from_header(event.headers);
+    const topics = await User.findOne({username: username}, 'followed_topics');
+    return {
+        'statusCode': 200,
+        'body': JSON.stringify({
+            topics: topics
+        })
+    }
+};
+
+// TODO: Test this. Updates followed user.
+exports.follow_user = async (_, event) => {
+    if (event.queryStringParameters != null)
+        console.error("Parameters currently ignored");
+
+    // get current user
+    const username = await get_user_from_header(event.headers);
+    var user = await User.findOne({username: username});
+
+    // get query params for the username of the user to follow
+    const usernameToFollow = event.queryStringParameters.usernameToFollow;
+    user.following.push(usernameToFollow);
+    user.save();
+
+    return {
+        'statusCode': 200,
+        'body': JSON.stringify({
+            following: following
+        })
+    }
+};
+
+// TODO: Unfollow users.
+exports.unfollow_user = async (_, event) => {
```

Comment:
Should we put this under DELETE /follow?username=username

Reply:
I'll leave it under PATCH with follow for now. I think PATCH might be better since we're updating the array value.